hazelcast/src/main/java/com/hazelcast/nio/tcp/WriteHandler.java

```diff
@@ -93,15 +94,32 @@ private void createWriter(String protocol) {
                 outputBuffer.put(stringToBytes(Protocols.CLUSTER));
                 registerOp(SelectionKey.OP_WRITE);
             } else if (Protocols.CLIENT_BINARY.equals(protocol)) {
+                adjustBufferSizesForClientConnection();
                 socketWriter = new SocketClientDataWriter();
             } else if (Protocols.CLIENT_BINARY_NEW.equals(protocol)) {
+                adjustBufferSizesForClientConnection();
                 socketWriter = new SocketClientMessageWriter();
             } else {
+                adjustBufferSizesForClientConnection();
                 socketWriter = new SocketTextWriter(connection);
             }
         }
     }
 
+    private void adjustBufferSizesForClientConnection()  {
+        if (connectionManager.socketClientSendBufferSize == connectionManager.socketSendBufferSize) {
+            return;
+        }
+
+        outputBuffer = ByteBuffer.allocate(connectionManager.socketClientSendBufferSize);
+        try {
+            connection.getSocketChannelWrapper().socket().setSendBufferSize(connectionManager.socketClientSendBufferSize);
```

Comment:
I prefer giving responsibility of setting buffer sizes to `TcpIpConnection` class by introducing new two methods instead of this long call.
- `setReceiveBufferSize(int size)`
- `setSendBufferSize(int size)`

So inside itself, it can call `socketChannel.socket().setReceiveBufferSize(size)` and `socketChannel.socket().setSendBufferSize(size)`